frontend/__tests__/src/pages/ProjectDetails.test.tsx

```diff
@@ -60,19 +65,191 @@ describe('ProjectDetailsPage Component', () => {
     )
 
     render(<ProjectDetailsPage />)
-
     await waitFor(() => {
       expect(screen.getByText('user1')).toBeInTheDocument()
     })
   })
 
   test('renders project URL as clickable link', async () => {
     render(<ProjectDetailsPage />)
-
     await waitFor(() => {
       const link = screen.getByText('https://example.com')
       expect(link.tagName).toBe('A')
       expect(link).toHaveAttribute('href', 'https://example.com')
     })
   })
+
+  // Below are additional tests to cover toggling “Show more/Show less” for languages, topics, and contributors.
```

Comment:
Please avoid adding excessive comments. You should comment only difficult to understand code.